### PR TITLE
Make requires lazy to improve startup times

### PR DESCRIPTION
**Pull request by trentmwillis**
Addressing #122. ~~Added some simple caching as well, not sure if it's beneficial or not.~~ These changes seem to cut startup times by about 50% on average.

cc @stefanpenner

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -48,8 +46,27 @@ module.exports = {
     return path.join(__dirname, 'blueprints');
   },
 
+  require: function(name) {
+    if (!requireCache) {
```
Comment (rwjblue) [inline index.js]: > Added some simple caching as well, not sure if it's beneficial or not

`require` caches by default, no need to do manual caching.

Comment (trentmwillis) [inline index.js]: TIL...

Comment (stefanpenner) [inline index.js]: https://nodejs.org/api/globals.html#globals_require_cache

**Comment by trentmwillis** [comment]
@rwjblue updated

**Comment by stefanpenner** [comment]
:shipit:

**Pull request merged**